Common/Securities/Future/FuturesExpiryUtilityFunctions.cs

```diff
@@ -124,9 +125,14 @@ public static DateTime NthFriday(DateTime time, int n)
             }
 
             var daysInMonth = DateTime.DaysInMonth(time.Year, time.Month);
-            return (from day in Enumerable.Range(1, daysInMonth)
-                    where new DateTime(time.Year, time.Month, day).DayOfWeek == DayOfWeek.Friday
-                    select new DateTime(time.Year, time.Month, day)).ElementAt(n - 1);
+            var nthFriday = (from day in Enumerable.Range(1, daysInMonth)
+                             where new DateTime(time.Year, time.Month, day).DayOfWeek == DayOfWeek.Friday
+                             select new DateTime(time.Year, time.Month, day)).ElementAt(n - 1);
+
+            // If not trading day, get the previous day
+            while (!NotHoliday(nthFriday)) nthFriday = nthFriday.AddDays(-1);
```

Comment:
In my opinion, this function should return the third Friday, independently if it's holiday or not. The holiday case should be handled by each expiry function.